e2e_tests/tests/cluster/test_job_queue.py

```diff
@@ -13,27 +13,29 @@
 def test_job_queue_adjust_weight() -> None:
     config = conf.tutorials_path("mnist_pytorch/const.yaml")
     model = conf.tutorials_path("mnist_pytorch")
-    for _ in range(2):
-        exp.create_experiment(config, model)
+    exp_ids = [exp.create_experiment(config, model) for _ in range(2)]
 
-    jobs = JobInfo()
-    ok = jobs.refresh_until_populated()
-    assert ok
+    try:
```

Comment:
I don't understand what the try/finally block is intended for. I get from the commit notes that experiments needed to be killed, and that this does that. But I don't understand why continuing experiments made the test flaky, and I don't understand why you went with try/finally instead of just killing the experiments before returning outside of any try.

What was the problem with this test? The CI run linked to in the description has a failure like
> FAILED tests/command/test_shell.py::test_open_shell
not in `test_job_queue_adjust_weight` (which to me looks like it passes)

Reply:
```test_job_queue_adjust_weight``` passed but scheduled two experiments that take a relatively long time.

This is unnecessarily and caused the shell test to fail since it expected the shell to open in a reasonable amount of time. Instead the shell test timed out because the test cluster was busy was running these experiments. I made the shell test also more resilient to waiting on other jobs that ran.

It is in a finally clause since we shouldn't leak these experiments even if the test fails. We can't kill the experiments right after we start them since we need them to be running for the job queue test functions.

Reply:
Aha! Thanks! That all makes a ton of sense.

How do you like [pytest's xunit-style setup/teardowns](https://docs.pytest.org/en/7.3.x/how-to/xunit_setup.html#method-and-function-level-setup-teardown) for this? Maybe with a global to store the `exp_id`s? Is that gross / or overengineering?

I think it'd look like
```python
def setup_function(fun): 
    config = conf.tutorials_path("mnist_pytorch/const.yaml")
    model = conf.tutorials_path("mnist_pytorch")
    global exp_ids
    exp_ids = [exp.create_experiment(config, model) for _ in range(2)]

def teardown_function(fun):
    exp.kill_experiments(exp_ids)
```
I don't think it's necessary if you use the pytest framework, but lacking that, I think a comment describing the intent would be helpful.

Reply:
The teardown functions look pretty gross to me but that is just me.

I added a comment explaining intent instead.